src/vcstools/git.py

```diff
@@ -340,14 +340,18 @@ def get_current_version_label(self):
             return '<detached>'
         result = branch
         remote_branch = self.get_branch_parent(allow_other_remote=True)
-        if (remote_branch is not None and remote_branch != branch):
-            result += ' < ' + remote_branch
+        if remote_branch is not None:
+            if remote_branch.startswith('origin/'):
+                remote_branch = remote_branch[len('origin/'):]
+            if remote_branch != branch:
+                result += ' < ' + remote_branch
         return result
 
-    def get_remote_version(self, spec=None, fetch=False):
-        parent_branch = self.get_branch_parent(fetch=fetch)
+    def get_remote_version(self, fetch=False, allow_other_remote=True):
```

Comment:
it's not good to extend the public API with a parameter for one VCS. I would create an internal API like __do_get_remote_version, and keep the public method same as all others.